DeviceSimulation/SimulationTests.cs

```diff
@@ -29,5 +40,190 @@ public void Should_Return_OK_Status()
+                    } 
+                ]
+            }");
+            var request = new HttpRequest(DS_ADDRESS + "/simulations");
+            request.AddHeader("Content-Type", "application/json");
+            request.SetContent(simulation);
+            var response = this.httpClient.PostAsync(request).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Thread.Sleep(15000);
+
+            ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(IOTHUB_CONNECTION_STRING);
+            Task<CloudToDeviceMethodResult> directResponseFuture = serviceClient.InvokeDeviceMethodAsync("truck-01.0",
+                new CloudToDeviceMethod("FirmwareUpdate"));
+            CloudToDeviceMethodResult response1 = await directResponseFuture.ConfigureAwait(false);
+            Assert.Equal(200, response1.Status);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public async void Should_Reboot_SimulatedDevice()
+        {
+            //delete the simulation that is already running
+            this.Should_Delete_Existing_Simulation();
+            var simulation = JObject.Parse(@"{   
+                'Enabled': true, 
+                'DeviceModels': [   
+                    {   
+                        'Id': 'chiller-01', 
+                        'Count': 5 
+                    } 
+                ]
+            }");
+            var request = new HttpRequest(DS_ADDRESS + "/simulations");
+            request.AddHeader("Content-Type", "application/json");
+            request.SetContent(simulation);
+            var response = this.httpClient.PostAsync(request).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Thread.Sleep(15000);
+
+            ServiceClient serviceClient = ServiceClient.CreateFromConnectionString(IOTHUB_CONNECTION_STRING);
+            Task<CloudToDeviceMethodResult> directResponseFuture = serviceClient.InvokeDeviceMethodAsync("chiller-01.0",
+                new CloudToDeviceMethod("Reboot"));
+            CloudToDeviceMethodResult response1 = await directResponseFuture.ConfigureAwait(false);
+
+            Assert.Equal(200, response1.Status);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Start_Given_Simulation()
+        {
+
+            var request1 = new HttpRequest(DS_ADDRESS + "/simulations/1");
+            var response1 = this.httpClient.GetAsync(request1).Result;
+            Assert.Equal(HttpStatusCode.OK, response1.StatusCode);
+            JObject JsonResponse = JObject.Parse(response1.Content);
+
+            string ETag = (string)JsonResponse["ETag"];
+
+            var simulation2 = "{" + $"'ETag': '{ETag}' ,'Enabled': true" + "}";
+            var byteArray = Encoding.ASCII.GetBytes(simulation2);
+
+            var request2 = (HttpWebRequest)WebRequest.Create("http://localhost:9003/v1/simulations/1");
+            request2.Method = "PATCH";
```

Comment:
suggest adding a private helper method for this request since it is reused in multiple tests